From 38afbd4f03a7b1816b7da30e7c5f5f9eb167f222 Mon Sep 17 00:00:00 2001
From: mzz2017 <mzz@tuta.io>
Date: Sun, 11 Dec 2022 19:17:20 +0800
Subject: [PATCH] fix: check the transport type to avoid abnormal exit of core

---
 service/core/serverObj/v2ray.go | 2 ++
 1 file changed, 2 insertions(+)

--- a/core/serverObj/v2ray.go
+++ b/core/serverObj/v2ray.go
@@ -333,6 +333,8 @@ func (v *V2Ray) Configuration(info Prior
 					Path: v.Path,
 				}
 			}
+		default:
+			return Configuration{}, fmt.Errorf("unexpected transport type: %v", v.Net)
 		}
 		if strings.ToLower(v.TLS) == "tls" {
 			core.StreamSettings.Security = "tls"
